lib/jobs/switch-node-relations.js

```diff
@@ -139,10 +140,42 @@ function switchRelationsJobFactory(
                 return waterline.nodes.updateByIdentifier(
                     lldpData.node,
                     { relations: relations });
+            })
+            .then(function () {
+                return self.updateSwitchNodeRelations(lldpData.node);
             });
 
         };
 
+    SwitchRelationsJob.prototype.updateSwitchNodeRelations =
+        function updateSwitchNodeRelations(connectedNodeId) {
+            var self = this;
+            var switchNode;
+            return waterline.nodes.needByIdentifier(self.nodeId)
+            .then(function (node) {
+                switchNode = node;
+                return switchNode.relations;
+            })
+            .filter(function (entry) {
+                //return entries except the one with relation type = powers
+                return (entry.relationType !== SWITCH_RELATION_TYPE);
+            })
+            .then(function (relations) {
+                //update Switch relation type powers
+                var targets = _getTargets(switchNode, SWITCH_RELATION_TYPE);
+                if (targets.indexOf(connectedNodeId) < 0) {
+                    targets.push(connectedNodeId);
+                }
+                relations.push({
```

Comment:
I haven't see the code to delete the out-of-date switch relation, if so, there may be duplicated relations in the node document. If you have handle this, please tell me the code, otherwise, please fix it.